cloudformation/alarms.yaml

```diff
@@ -491,3 +491,93 @@ Resources:
       Period: 60
       Statistic: Maximum
       Threshold: 0
+  # cloudwatch synthetic support
+  MetadataCanary:
+    Type: AWS::Synthetics::Canary
+    Properties:
+      ArtifactS3Location: !Join ['', ['s3://', !Ref SyntheticsBucket, '/', !Join ['', [!Ref Stage, ig, canary]]]]
+      Code:
+        Handler: 'index.handler'
+        Script: |

```

Comment:
Could you please try using s3 bucket to store the code and [reference it](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-synthetics-canary-code.html) directly? It'll be much cleaner for the CF template and will set up a great example to follow for other canary alarms to come in the future. 

You can follow the example [here](https://github.com/awslabs/fhir-works-on-aws-deployment/blob/mainline/bulkExport/uploadGlueScriptsToS3.ts) to upload a script to S3 bucket as part of the build.

Reply:
Thanks, I'll take a look.

Reply:
I'm probably not going to get time for this in awhile. It looks like it's a bit of work to make the custom cloudformation resource  generic enough to handle any synthetic script dropped into a dir.